### PR TITLE
feat: ensemble scoring

### PR DESCRIPTION
**Pull request by ab93**
- Enable ensemble scoring
- Add static_threshold_wt field in config
- Increase min training datapoints to 1000 -> 2000
- Increase threshold factor 3 -> 5 to make the model less sensitive
- Use TanhScaler as default preprocessing step

### CODE REVIEW

**Review by s0nicboOm — COMMENTED** [review]
LGTM!

File: numaprom/udf/postprocess.py [hunk 1]
```diff
@@ -147,25 +148,103 @@ def _publish(final_score: float, payload: StreamPayload) -> List[bytes]:
...
+
+    # Perform score normalization
+    postproc_clf = TanhNorm()
+    norm_static_score = postproc_clf.transform(mean_static_score)
+    norm_model_score = postproc_clf.transform(mean_model_score)
+
+    # Calculate ensemble score
+    static_wt = metric_config.static_threshold_wt
+    model_wt = 1.0 - metric_config.static_threshold_wt
+    ensemble_score = (static_wt * norm_static_score) + (model_wt * norm_model_score)
```
Comment (nkoppisetty) [inline numaprom/udf/postprocess.py:174]: Also, if you are setting default static weight to 0, we would not need separate methods to calculate scores right?

File: numaprom/udf/postprocess.py [hunk 2]
```diff
@@ -147,25 +148,103 @@ def _publish(final_score: float, payload: StreamPayload) -> List[bytes]:
...
+            "%s - Final static threshold score: %s, keys: %s",
+            payload.uuid,
+            final_score,
+            payload.composite_keys,
+        )
 
-    postproc_clf = TanhNorm()
-    norm_score = postproc_clf.transform(raw_mean_score)
+    # Compute ensemble score if there is a valid weight for static threshold
+    elif metric_config.static_threshold_wt:
```
Comment (nkoppisetty) [inline numaprom/udf/postprocess.py:227]: you can just call _calculate_ensemble_score method in else

File: numaprom/udf/postprocess.py [hunk 3]
```diff
@@ -147,25 +148,103 @@ def _publish(final_score: float, payload: StreamPayload) -> List[bytes]:
...
+        payload.uuid,
+        norm_model_score,
+        norm_static_score,
+        static_wt,
+    )
+
+    return ensemble_score
+
+
+def _calculate_score(payload: StreamPayload) -> float:
```
Comment (nkoppisetty) [inline numaprom/udf/postprocess.py:187]: remove this

**Review by nkoppisetty — CHANGES_REQUESTED** [review]

**Review by nkoppisetty — APPROVED** [review]

**Pull request merged**